python/pyspark/sql/connect/dataframe.py

```diff
@@ -1147,18 +1147,28 @@ def collect(self) -> List[Row]:
         table = self._session.client.to_table(query)
 
         rows: List[Row] = []
-        for row in table.to_pylist():
-            _dict: Dict[Any, Any] = {}
-            for k, v in row.items():
+        i = 0
+        while i < table.num_rows:
+            keys: List[Any] = []
+            values: List[Any] = []
+            j = 0
+            while j < table.num_columns:
+                k = table.column_names[j]
+                v = table.column(j).to_pylist()[i]
```

Comment:
is it possible that `table.column(j).to_pylist()` also have this issue?

`table.column(j)` could also be a struct type

Reply:
table.column(j) is instance of `pyarrow.ChunkedArray`, but  table is instance of `pyarrow.Table`

Reply:
```
>>> import pyarrow as pa
>>> n_legs = pa.chunked_array([[2, 2, 4], [4, None, 100]])
>>> n_legs.to_pylist()
[2, 2, 4, 4, None, 100]
```

pyarrow.ChunkedArray doesn't  have schema.

Reply:
`pyarrow.ChunkedArray` does not hit this issue?

Reply:
got it